### PR TITLE
Fix for two build errors with CMake

### PR DESCRIPTION
**Pull request by Semphriss**
## Bug 1: Position independent code

Compiling SDL2_image as a shared library will not enable CMAKE_POSITION_INDEPENDENT_CODE, which is necessary for libjpeg as it is always compiled as a static library.

Before this fix, attempting to compile SDL2_image with BUILD_SHARED_LIBS=ON gives the following message at compile time:
```
/usr/bin/ld: external/jpeg-9d/libjpeg.a(jcapimin.c.o): relocation R_X86_64_PC32 against symbol `jpeg_natural_order' can not be used when making a shared object; recompile with -fPIC
```

I fixed this by enabling CMAKE_POSITION_INDEPENDENT_CODE (which is CMake's flag for -fPIC) for all builds, regardless of BUILD_SHARED_LIBS.

## Bug 2: Compiling with system SDL

SDL2_image attempts to compile against `SDL2::SDL2` all the time, although that target may not be valid if SDL was found with `find_package(SDL2 REQUIRED)`. The variables `SDL2_LIBRARIES` and `SDL2_LIBRARIES` serve that purpose; I added a check for `SDL2_FOUND` (which is truthy if and only if `find_package(SDL2 REQUIRED)` was used successfully) and made it use those variables instead.

**Note:** I noticed SDL2_image also uses `-DDLL_EXPORT` for shared libraries on Windows; since I don't have a Windows machine nearby at the moment, I couldn't check whether it was related to the was SDL2 is compiled or not, and I didn't add it in this PR. Testing dynamic linking on Windows might be necessary before merging this PR.

### CODE REVIEW

**Comment by sezero** [comment]
> I fixed this by enabling CMAKE_POSITION_INDEPENDENT_CODE

Wrong thing to do. The right thing would be including all external lib sources in the
SDL_image cmake script instead of add_subdirectory

**Comment by Semphriss** [comment]
Libjpeg always compile as a static library, as its CMakeLists.txt file says; an alternative to above would be to set `POSITION_INDEPENDENT_CODE` to only libjpeg through a patch.

**Comment by sezero** [comment]
> Libjpeg always compile as a static library, as its CMakeLists.txt file says; 

That one is not from mainstream libjpeg: it is a makeshift cmake script which was added originally by https://github.com/libsdl-org/SDL_image/commit/8fe59c40898f8e935d337ebba9867d8b3e76c317 and was intended only for android builds, so it is as much as incomplete/immature as SDL_image's own cmake script, if not more. I don't know whether there would be any fallout from such changes.

**Comment by Semphriss** [comment]
I've updated this PR to move the `-fPIC`-related code to the libjpeg folder. I've noticed there are also patches, but since the patch does not include the CMakeLists.txt, I put my code directly in the folder. I left this PR open for changes by maintainers so edits can be directly pushed to it if necessary.

File: CMakeLists.txt [hunk 1]
```diff
@@ -20,7 +20,6 @@ if (APPLE)
 	option(ENABLE_APPLE_IMAGEIO "Use native Mac OS X frameworks for loading images" ON)
 endif()
 
-# FIXME: ????
```
Comment (sezero) [inline CMakeLists.txt:23]: This is still a FIXME: I suggest not removing the note.

Comment (Semphriss) [reply CMakeLists.txt]: My bad - I forgot to re-add it when reverting my changes.

**Comment by slouken** [comment]
Can you review your changes with respect to the latest SDL_image source changes?
FYI, the jpeg library is now a separate repo and you can issue pull requests against that: https://github.com/libsdl-org/jpeg

**Comment by ccawley2011** [comment]
> For my personal knowledge, is there any downside to activating `CMAKE_POSITION_INDEPENDENT_CODE` for shared libraries? It seems, according to the [docs](https://cmake.org/cmake/help/latest/prop_tgt/POSITION_INDEPENDENT_CODE.html), that it is already true for shared libraries anyways.

Some platforms (RISC OS, PSP, possibly others) often have trouble when using PIC outside of shared libraries. Would it be possible to set this conditionally based on whether or not SDL2_image itself is built as a shared library?

File: CMakeLists.txt [hunk 2]
```diff
@@ -98,15 +98,20 @@ add_library(SDL2::image ALIAS SDL2_image)
...
-	  target_sources(SDL2_image PRIVATE version.rc)
+if(SDL2_FOUND)
+	target_link_libraries(SDL2_image PUBLIC ${SDL2_LIBRARIES})
+	target_include_directories(SDL2_image PUBLIC ${SDL2_INCLUDE_DIRS})
+else(SDL2_FOUND)
+	if (BUILD_SHARED_LIBS)
+		if(WIN32)
+		  target_compile_definitions(SDL2_image PRIVATE -DDLL_EXPORT)
+      target_sources(SDL2_image PRIVATE version.rc)
+		endif()
```
Comment (ccawley2011) [inline CMakeLists.txt:109]: This bit should be moved outside of the `SDL2_FOUND` check.

**Comment by sezero** [comment]
Is this still needed?

**Comment by sezero** [comment]
Close this in favor of #238 ?

**Comment by madebr** [comment]
I think current master already has fixes for the issues mentioned by OP.
@Semphriss can you confirm?

**Comment by sezero** [comment]
Looks like the OP lost interest..

**Comment by sezero** [comment]
> Looks like the OP lost interest..

Closing, because no response from OP.

**Pull request closed without merging**